traffic_ops/app/db/reencrypt/reencrypt.go

```diff
@@ -0,0 +1,406 @@
+		if err != nil {
+			fmt.Println("reading SSL Keys: ", err)
+			return err
+		}
+
+		reencryptedKeys, err := postgres.AesEncrypt(jsonKeys, newKey)
+		if err != nil {
+			fmt.Println("encrypting SSL Keys with new key: ", err)
+			return err
+		}
+
+		res, err := updateTx.Exec(`UPDATE sslkey SET data = $1 WHERE id = $2`, []byte(reencryptedKeys), id)
+		if err != nil {
+			fmt.Println("updating SSL Keys for id ", id, ": ", err)
+			return err
+		}
+		rowsAffected, err := res.RowsAffected()
+		if err != nil {
+			fmt.Println("error determining rows affected for reencrypting SSL Keys with id ", id)
+			return errors.New(fmt.Sprintf("determining rows affected for reencrypting SSL Keys with id %d", id))
+		}
+		if rowsAffected == 0 {
+			fmt.Println("no rows updated for reencrypting SSL Keys for id ", id)
+			return errors.New(fmt.Sprintf("no rows updated for reencrypting SSL Keys for id %d", id))
+		}
+
+	}
+
+	return nil
+}
+
+func reEncryptUrlSigKeys(db *sqlx.DB, previousKey []byte, newKey []byte) error {
+	tx, err := db.Begin()
+	if err != nil {
+		fmt.Println("transaction begin failed ", err, " ", tx)
+		os.Exit(0)
+	}
+	defer tx.Commit()
+
+	rows, err := tx.Query("SELECT deliveryservice, data FROM url_sig_key")
+	if err != nil {
+		fmt.Println("querying: ", err)
+		os.Exit(0)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		updateTx, err := db.Begin()
+		if err != nil {
+			fmt.Println("transaction begin failed ", err, " ", updateTx)
+			os.Exit(0)
+		}
+		defer updateTx.Commit()
+
+		ds := ""
+		var encryptedSslKeys []byte
+		if err = rows.Scan(&ds, &encryptedSslKeys); err != nil {
+			fmt.Println("getting URL Sig Keys: ", err)
+			return err
+		}
+		jsonKeys, err := postgres.AesDecrypt(encryptedSslKeys, previousKey)
+		if err != nil {
+			fmt.Println("reading URL Sig Keys: ", err)
+			return err
+		}
+
+		reencryptedKeys, err := postgres.AesEncrypt(jsonKeys, newKey)
+		if err != nil {
+			fmt.Println("encrypting URL Sig Keys with new key: ", err)
+			return err
+		}
+
+		res, err := updateTx.Exec(`UPDATE url_sig_key SET data = $1 WHERE deliveryservice = $2`, []byte(reencryptedKeys), ds)
+		if err != nil {
+			fmt.Println("updating URL Sig Keys for deliveryservice ", ds, ": ", err)
+			return err
+		}
+		rowsAffected, err := res.RowsAffected()
+		if err != nil {
+			fmt.Println("error determining rows affected for reencrypting URL Sig Keys with deliveryservice ", ds)
+			return errors.New(fmt.Sprintf("determining rows affected for reencrypting URL Sig Keys with deliveryservice %d", ds))
+		}
+		if rowsAffected == 0 {
+			fmt.Println("no rows updated for reencrypting URL Sig Keys for id ", ds)
+			return errors.New(fmt.Sprintf("no rows updated for reencrypting URL Sig Keys for deliveryservice %d", ds))
+		}
+
+	}
+
+	return nil
+}
+
+func reEncryptUriSigningKeys(db *sqlx.DB, previousKey []byte, newKey []byte) error {
+	tx, err := db.Begin()
+	if err != nil {
+		fmt.Println("transaction begin failed ", err, " ", tx)
+		os.Exit(0)
```

Comment:
Same as above RE: exit with failure code on failure and also returning an error versus forcibly quitting